windows/bleachbit.nsi

```diff
@@ -17,163 +17,731 @@
+; !insertmacro MUI_UNPAGE_DIRECTORY
+!insertmacro MUI_UNPAGE_COMPONENTS
+!insertmacro MUI_UNPAGE_INSTFILES
+!define MUI_UNFINISHPAGE_NOAUTOCLOSE
+!insertmacro MUI_UNPAGE_FINISH
+
+; MUI_LANGUAGE[EX] should be inserted after the MUI_[UN]PAGE_* macros!
 
 
 ;--------------------------------
-;Language Selection Dialog Settings
+;Interface Settings
 
-  ;Remember the installer language
-  !define MUI_LANGDLL_REGISTRY_ROOT "HKCU"
-  !define MUI_LANGDLL_REGISTRY_KEY "Software\${prodname}"
-  !define MUI_LANGDLL_REGISTRY_VALUENAME "Installer Language"
+; Show a message box with a warning when the user wants to close the installer:
+!define MUI_ABORTWARNING
+!define MUI_ABORTWARNING_CANCEL_DEFAULT
+!define MUI_UNABORTWARNING
+!define MUI_UNABORTWARNING_CANCEL_DEFAULT
 
+; Show all languages, despite user's codepage:
+; https://nsis.sourceforge.io/Why_does_the_language_selection_dialog_hide_some_languages
+!define MUI_LANGDLL_ALLLANGUAGES
 
-;--------------------------------
-;Pages
 
-; installer
-  !insertmacro MUI_PAGE_LICENSE "..\COPYING"
-  !insertmacro MULTIUSER_PAGE_INSTALLMODE
-  !insertmacro MUI_PAGE_DIRECTORY
-  !insertmacro MUI_PAGE_COMPONENTS
-  !insertmacro MUI_PAGE_INSTFILES
+;--------------------------------
+;Language Selection Dialog Settings
 
-  !define MUI_FINISHPAGE_NOAUTOCLOSE
-  !define MUI_FINISHPAGE_RUN "$INSTDIR\${prodname}.exe"
-  !define MUI_FINISHPAGE_LINK "Visit the ${prodname} web site"
-  !define MUI_FINISHPAGE_LINK_LOCATION "https://www.bleachbit.org"
-  !insertmacro MUI_PAGE_FINISH
+; Better not doing that, or the user can never ever change his language!
+; OK, we give it with the new code a try again... ^^
 
-; uninstaller
-  !insertmacro MULTIUSER_UNPAGE_INSTALLMODE
-  !insertmacro MUI_UNPAGE_CONFIRM
-  !insertmacro MUI_UNPAGE_INSTFILES
+; Remember the installer language
+!define MUI_LANGDLL_REGISTRY_ROOT "HKCU"
+!define MUI_LANGDLL_REGISTRY_KEY "Software\${prodname}"
+!define MUI_LANGDLL_REGISTRY_VALUENAME "Installer Language"
 
 
 ;--------------------------------
 ;Languages
 
-  !insertmacro MUI_LANGUAGE "English"
+; MUI_LANGUAGE[EX] should be inserted after the MUI_[UN]PAGE_* macros!
+
+; Languages additionaly available in bleachbit_lang.nsh and NsisMultiUserLang.nsh are in comments
+!insertmacro MUI_LANGUAGE "English"
 !ifndef NoTranslations
+; !insertmacro MUI_LANGUAGE "AFRIKAANS"
   !insertmacro MUI_LANGUAGE "Albanian"
   !insertmacro MUI_LANGUAGE "Arabic"
   !insertmacro MUI_LANGUAGE "Armenian"
   !insertmacro MUI_LANGUAGE "Asturian"
   !insertmacro MUI_LANGUAGE "Basque"
   !insertmacro MUI_LANGUAGE "Belarusian"
   !insertmacro MUI_LANGUAGE "Bosnian"
+; !insertmacro MUI_LANGUAGE "BRETON"
   !insertmacro MUI_LANGUAGE "Bulgarian"
   !insertmacro MUI_LANGUAGE "Catalan"
+; !insertmacro MUI_LANGUAGE "CORSICAN"
   !insertmacro MUI_LANGUAGE "Croatian"
   !insertmacro MUI_LANGUAGE "Czech"
   !insertmacro MUI_LANGUAGE "Danish"
   !insertmacro MUI_LANGUAGE "Dutch"
+; !insertmacro MUI_LANGUAGE "ESPERANTO"
```

Comment:
Please include translations only when there is a [BleachBit translation in Launchpad](https://translations.launchpad.net/bleachbit/trunk/+pots/bleachbit/) that is at least partially complete for two reasons

1. It increases the size of the NSIS installer
2. It ultimately does not give a good experience to the user to be able to install the software but not operate the application

What is the minimum translation? Maybe 5% or 10%. For example,  Kazakh is 2% complete, so I wouldn't provide an NSIS translation.

Reply:
It was just a comment and not included! Also: This comment was mainly for my future me to know which languages I can use with MUI_LANGUAGE... See my comment on top of this section in the code and my comment beyond here in the issue...

Reply:
Addition: I added now the languages that get provided with BleachBit already!
The users should IMHO be able to install BB in this language, they also use in BB.
So "Farsi" is e.g. "Persian" and "PortugueseBR" is "Basilian Portuguese"...